cmake/json_test.cmake

```diff
@@ -4,7 +4,21 @@ set(_json_test_cmake_list_file ${CMAKE_CURRENT_LIST_FILE})
 # download test data
 #############################################################################
 
-include(download_test_data)
+# if variable is set, use test data from given directory rather than downloading them
+if(JSON_TEST_DATA_DIRECTORY)
+    add_custom_target(download_test_data)
+    file(WRITE ${CMAKE_BINARY_DIR}/include/test_data.hpp "#define TEST_DATA_DIRECTORY \"${JSON_TEST_DATA_DIRECTORY}\"\n")
+else()
+    find_package(Git REQUIRED)
+    # target to download test data
+    add_custom_target(download_test_data
+        COMMAND test -d json_test_data || ${GIT_EXECUTABLE} clone -c advice.detachedHead=false --branch v${JSON_TEST_DATA_VERSION} ${JSON_TEST_DATA_URL}.git --quiet --depth 1
```

Comment:
For follow-up PR: `.git` should be part of the URL variable value so other URLs can be used.